src/main/java/com/meetup/server/auth/support/handler/OAuth2LoginSuccessHandler.java

```diff
@@ -39,13 +39,48 @@ public void onAuthenticationSuccess(
         cookieUtil.setAccessTokenCookie(response, accessToken);
         cookieUtil.setRefreshTokenCookie(response, refreshToken);
 
-        String targetUrl = createRedirectUrlWithTokens();
+        String targetUrl = createRedirectUrlWithTokens(request);
         getRedirectStrategy().sendRedirect(request, response, targetUrl);
     }
 
-    private String createRedirectUrlWithTokens() {
-        return UriComponentsBuilder.fromUriString(successRedirectUri)
+    private String createRedirectUrlWithTokens(HttpServletRequest request) {
+        log.info("[Redirect URI] - {}", createRedirectUrl(request));
+        return UriComponentsBuilder.fromUriString(createRedirectUrl(request))
                 .build()
                 .toUriString();
     }
+
+    private String createRedirectUrl(HttpServletRequest request) {
+        String state = request.getParameter("state");
+        String to = null;
+        String eventId = null;
+
+        if (state != null && !state.isBlank()) {
+            String decodedState = java.net.URLDecoder.decode(state, java.nio.charset.StandardCharsets.UTF_8);
+
+            for (String param : decodedState.split("&")) {
+                String[] keyValue = param.split("=", 2);
+                if (keyValue.length == 2) {
+                    if ("to".equals(keyValue[0])) {
+                        to = keyValue[1];
+                    } else if ("eventId".equals(keyValue[0])) {
+                        eventId = keyValue[1];
+                    }
+                }
+            }
+        }
+
+        String redirectBase;
+        redirectBase = successRedirectUri + "/oauth/kakao/callback";
```

Comment:
```suggestion
        String redirectBase = successRedirectUri + "/oauth/kakao/callback";
```
변수 초기화할 때 합치는게 좋을 거 같아요!